core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/ExcelInputReader.java

```diff
@@ -309,7 +309,7 @@ private void createDatasets(@Nullable PipelineConfigurer pipelineConfigurer, @Nu
 
         if (pipelineConfigurer != null) {
           pipelineConfigurer.createDataset(excelInputreaderConfig.errorDatasetName, Table.class, datasetProperties);
-        } else if (context != null) {
+        } else if (context != null && !context.datasetExists(excelInputreaderConfig.errorDatasetName)) {
           context.createDataset(excelInputreaderConfig.errorDatasetName, Table.class.getName(), datasetProperties);
```

Comment:
Lgtm.

Please file a jira, since I don't think user should have to protect against
already exists.
Like in app configure, they simply call create dataset.

On Aug 20, 2016 10:36 AM, "sagarkapare" <notifications@github.com> wrote:

> In core-plugins/src/main/java/co/cask/hydrator/plugin/batch/
> source/ExcelInputReader.java
> <https://github.com/caskdata/hydrator-plugins/pull/419#discussion_r75582142>
> :
>
> > @@ -309,7 +309,7 @@ private void createDatasets(@Nullable PipelineConfigurer pipelineConfigurer, @Nu
> >
> >          if (pipelineConfigurer != null) {
> >            pipelineConfigurer.createDataset(excelInputreaderConfig.errorDatasetName, Table.class, datasetProperties);
> > -        } else if (context != null) {
> > +        } else if (context != null && !context.datasetExists(excelInputreaderConfig.errorDatasetName)) {
> >            context.createDataset(excelInputreaderConfig.errorDatasetName, Table.class.getName(), datasetProperties);
>
> We create datasets either during configure time or during runtime
> depending on the field is macro or not. In this case there is a logical
> error in which if field is not macro then we were creating dataset in
> configure time as well as in runtime which causes MapReduce to fail. So fix
> is to do not attempt the creation at runtime if dataset already created
> during configure time.
> Similar checks are done at other places https://github.com/caskdata/
> hydrator-plugins/pull/409/files#diff-a2609c590a55b09cdb5f0132ea8f14a4R119
> but was missing here.
>
> I think this logic can be further simplified but can be done as separate
> PR (post 3.5).
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/caskdata/hydrator-plugins/pull/419/files/5548a0de3ebcab917eb23e6b350d092fb81f2ff8#r75582142>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACU_Eb-kyGV3GlaGLZyi8y-Ak31IMAc0ks5qhzsBgaJpZM4JpHfg>
> .
>